### PR TITLE
Switch async images to UBI

### PR DESCRIPTION
**Pull request by tomgeorge**
Switch the alpine images to UBI,  #10 

Signed-off-by: Tom George <tg82490@gmail.com>

### CODE REVIEW

**Review by ibuziuk — COMMENTED** [review]
@tomgeorge could you please clarify how the PR was tested, also wondering it there are any issues with the following config / scripts after the UBI switch:
- https://github.com/che-incubator/workspace-data-sync/blob/master/dockerfiles/storage/sshd_config
- https://github.com/che-incubator/workspace-data-sync/tree/master/dockerfiles/agent/scripts
- https://github.com/che-incubator/workspace-data-sync/blob/master/dockerfiles/agent/cron/backup-cron-job

File: dockerfiles/agent/Dockerfile [hunk 1]
```diff
@@ -20,36 +21,35 @@ ENV SUPERCRONIC_URL=https://github.com/aptible/supercronic/releases/download/v0.
     SUPERCRONIC_SHA1SUM=5ddf8ea26b56d4a7ff6faecdd8966610d5cb9d85
 
 COPY cron/backup-cron-job /etc/crontabs/backup-cron-job
-COPY scripts scripts
+COPY scripts /scripts
+COPY content_sets_centos8.repo /etc/yum.repos.d/
 
 # Add user that will be able to start watcher binary but nothing more
 # the result will be propagated then into scratch image
 # See https://stackoverflow.com/a/55757473/12429735RUN
```
Comment (ibuziuk) [inline dockerfiles/agent/Dockerfile]: probably the comment should be udpated

**Review by benoitf — DISMISSED** [review]
AFAIK ubi and non-ubi are separated Dockerfiles

like Dockerfile with alpine and Dockerfile.rhel with ubi8 from

**Comment by ibuziuk** [comment]
@benoitf we do not plan to have non-ubi dockerfiles

**Comment by benoitf** [comment]
@ibuziuk why ? all images of che are not ubi based by default

**Comment by ibuziuk** [comment]
@benoitf I believe it is up to plugin / component developers to decide ;-)
image puller is UBI based by default, eventhough we have centos image, it is not used anywhere (with the recent centos stream announcement it is doomed to be deprecated in general). In order to lower the support burden, I would opt for ubi / ubi-minimal only for async-storage

**Comment by benoitf** [comment]
@ibuziuk 
```
quay.io/eclipse/che-workspace-data-sync-storage   latest               7b0123559264   48 seconds ago       251MB
quay.io/eclipse/che-sidecar-workspace-data-sync   latest               565bce0cc468   About a minute ago   248MB
quay.io/eclipse/che-workspace-data-sync-storage   alpine               66abd0279cd9   3 minutes ago        12.3MB
quay.io/eclipse/che-sidecar-workspace-data-sync   alpine               b615ce49c122   3 minutes ago        39.1MB
```

it's a diff of more than 200MB per image for nothing
Images should be as small as possible for upstream che.

**Comment by ibuziuk** [comment]
@tomgeorge maybe it is possible to use - https://catalog.redhat.com/software/containers/ubi8/ubi-minimal/5c359a62bed8bd75a2c3fba8 
@benoitf we can provide UBI & non-UBI versions. However, managing extra images creates an additional support burden for the teams and I believe we need to find an alternative to alpine that would be also suitable for product

**Comment by tomgeorge** [comment]
I have working `ubi-minimal` images but they are still pretty big:

```
quay.io/eclipse/che-sidecar-workspace-data-sync                                    latest         155288cd5b71   11 minutes ago      173MB
quay.io/eclipse/che-workspace-data-sync-storage                                    latest         4f1236fc9180   About an hour ago   163MB
```

**Review by ibuziuk — COMMENTED** [review]
@tomgeorge based on the discussion with @benoitf let's have separate ubi and alpine images for the product and project respectively

**Comment by tomgeorge** [comment]
Added the alpine Dockerfiles back and named the ubi dockerfiles `ubi.Dockerfile`

**Review by ibuziuk — APPROVED** [review]

**Comment by ibuziuk** [comment]
@benoitf could you please review?

File: dockerfiles/agent/ubi.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+#
+# Copyright (c) 2019-2020 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+
+FROM registry.redhat.io/ubi8/ubi-minimal:8.3
```
Comment (benoitf) [inline dockerfiles/agent/ubi.Dockerfile]: Please add a comment line
```
# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
```
Helping to see new images

Also, please stick to a given version like `FROM registry.access.redhat.com/ubi8-minimal:8.3-230`

File: dockerfiles/agent/ubi.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
```
Comment (benoitf) [inline dockerfiles/agent/ubi.Dockerfile]: I think it's a copy/error typo of the header (Dockerfile is not a bash script)

File: dockerfiles/agent/ubi.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+COPY cron/backup-cron-job /etc/crontabs/backup-cron-job
+COPY scripts /scripts
+#
+# Add user that will be able to start watcher binary but nothing more
+# the result will be propagated then into scratch image
+# See https://stackoverflow.com/a/55757473/12429735RUN
+#
+RUN microdnf update -y \ 
+      && microdnf install -y \
```
Comment (benoitf) [inline dockerfiles/agent/ubi.Dockerfile:32]: please follow convention of && at end of lines or beginning of lines

File: dockerfiles/storage/ubi.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+#
+# Copyright (c) 2019-2020 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+
+FROM registry.redhat.io/ubi8/ubi-minimal:8.3
```
Comment (benoitf) [inline dockerfiles/storage/ubi.Dockerfile]: same remarks than for the other Dockerfile (bash header, comment to see new updates and stick to a fine grain version tag)

File: dockerfiles/agent/ubi.Dockerfile (hunk 4 above)
Comment (tomgeorge) [reply dockerfiles/agent/ubi.Dockerfile:32]: I do have && at beginning of lines, it's the same as `dockerfiles/agent/Dockerfile`?

Comment (benoitf) [reply dockerfiles/agent/ubi.Dockerfile:32]: Ok maybe I read the only wrong line
```
&& mkdir  /var/run/sshd && \
```

**Review by benoitf — APPROVED** [review]

**Pull request merged**